### PR TITLE
[BEAM-13576] update bom

### PR DESCRIPTION
**Pull request by reuvenlax**
R: @yirutang

### CODE REVIEW

**Comment by reuvenlax** [comment]
Run Dataflow ValidatesRunner

**Comment by reuvenlax** [comment]
Run PostCommit_Java_Dataflow

**Comment by reuvenlax** [comment]
Run PostCommit_Java_DataflowV2

**Comment by reuvenlax** [comment]
Run SQL PostCommit

**Comment by reuvenlax** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Review by yirutang — APPROVED** [review]

**Comment by reuvenlax** [comment]
Run Java PreCommit

**Comment by reuvenlax** [comment]
Run PostCommit_Java_Dataflow

**Comment by reuvenlax** [comment]
Run PostCommit_Java_DataflowV2

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -456,7 +456,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def google_code_gson_version = "2.8.9"
     def google_oauth_clients_version = "1.32.1"
     // Try to keep grpc_version consistent with gRPC version in google_cloud_platform_libraries_bom
-    def grpc_version = "1.41.1"
+    def grpc_version = "1.42.1"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:459]: Good.  https://storage.googleapis.com/cloud-opensource-java-dashboard/com.google.cloud/libraries-bom/24.1.1/artifact_details.html

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 2]
```diff
@@ -472,7 +472,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def postgres_version = "42.2.16"
     def powermock_version = "2.0.9"
     // Try to keep protobuf_version consistent with the protobuf version in google_cloud_platform_libraries_bom
-    def protobuf_version = "3.18.1"
+    def protobuf_version = "3.19.1"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:475]: Good. https://storage.googleapis.com/cloud-opensource-java-dashboard/com.google.cloud/libraries-bom/24.1.1/artifact_details.html

**Review by suztomo — APPROVED** [review]
Looks good to me. Thank you for keeping the BOM updated.

**Comment by reuvenlax** [comment]
@suztomo I keep getting test failures that seem related to AVRO. Do we know if we have compatibility issues with newer versions of avro?

**Comment by suztomo** [comment]
In https://ci-beam.apache.org/job/beam_PostCommit_Java_DataflowV1_PR/58/console, I see NoClassDefFoundError :

```
12:39:49 > Task :runners:google-cloud-dataflow-java:coreSDKJavaLegacyWorkerIntegrationTest NO-SOURCE
12:39:53 
12:39:53 > Task :runners:google-cloud-dataflow-java:examplesJavaLegacyWorkerIntegrationTest
12:39:53 
12:39:53 org.apache.beam.examples.complete.AutoCompleteIT > initializationError FAILED
12:39:53     java.lang.NoClassDefFoundError at Class.java:-2
12:39:53         Caused by: java.lang.ClassNotFoundException at URLClassLoader.java:382
12:39:53 
12:39:53 org.apache.beam.examples.complete.TopWikipediaSessionsIT > initializationError FAILED
12:39:53     java.lang.NoClassDefFoundError at Class.java:-2
12:39:53         Caused by: java.lang.ClassNotFoundException at URLClassLoader.java:382
12:39:53 
12:39:53 org.apache.beam.examples.complete.TrafficRoutesIT > initializationError FAILED
12:39:53     java.lang.NoClassDefFoundError at Class.java:-2
12:39:53         Caused by: java.lang.ClassNotFoundException at URLClassLoader.java:382
12:39:53 
12:39:53 org.apache.beam.examples.complete.TrafficMaxLaneFlowIT > initializationError FAILED
12:39:53     java.lang.NoClassDefFoundError at Class.java:-2
12:39:53         Caused by: java.lang.ClassNotFoundException at URLClassLoader.java:382
12:39:53 
12:39:53 org.apache.beam.examples.complete.TfIdfIT > initializationError FAILED
12:39:53     java.lang.NoClassDefFoundError at Class.java:-2
12:39:53         Caused by: java.lang.ClassNotFoundException at URLClassLoader.java:382
12:39:53 
12:39:53 org.apache.beam.examples.cookbook.BigQueryTornadoesIT > initializationError FAILED
12:39:53     java.lang.NoClassDefFoundError at Class.java:-2
12:39:53         Caused by: java.lang.ClassNotFoundException at URLClassLoader.java:382
12:39:53 
12:39:53 org.apache.beam.examples.snippets.transforms.io.gcp.bigquery.BigQuerySamplesIT > initializationError FAILED
12:39:53     java.lang.NoClassDefFoundError at Class.java:-2
12:39:53         Caused by: java.lang.ClassNotFoundException at URLClassLoader.java:382
12:39:53 
12:39:53 7 tests completed, 7 failed
```

https://scans.gradle.com/s/sjhloabkaffks/tests/:runners:google-cloud-dataflow-java:examplesJavaLegacyWorkerIntegrationTest/org.apache.beam.examples.complete.TfIdfIT/initializationError?expanded-stacktrace=WyIwIiwiMC0xIl0&top-execution=1 shows `java.lang.NoClassDefFoundError: org/apache/beam/examples/complete/TfIdf$Options`.


What are the Avro problems you observe?

**Comment by reuvenlax** [comment]
Actually the problems all seem to be in PipelineOption instantiations.

**Comment by reuvenlax** [comment]
@lukecwik Have you ever seen errors like this? e.g. https://ci-beam.apache.org/job/beam_PostCommit_Java_DataflowV1_PR/58/testReport/junit/org.apache.beam.examples.cookbook/BigQueryTornadoesIT/initializationError/

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
I found https://github.com/apache/beam/pull/16388 "Fix class not found error for example integration tests", which was merged 6 days ago. The last run of this PR was before that merging. Retrying the postcommit integration tests.

**Comment by suztomo** [comment]
Run PostCommit_Java_Dataflow

**Comment by suztomo** [comment]
Run PostCommit_Java_DataflowV2

**Comment by suztomo** [comment]
They succeeded.

**Comment by lukecwik** [comment]
> They succeeded.

I think you are looking at the wrong run since I see that https://ci-beam.apache.org/job/beam_PostCommit_Java_DataflowV1_PR/58/ and https://ci-beam.apache.org/job/beam_PostCommit_Java_DataflowV2_PR/72/ failed.

**Comment by suztomo** [comment]
@lukecwik  I meant today's invocations:

- https://ci-beam.apache.org/job/beam_PostCommit_Java_DataflowV1_PR/63/
- https://ci-beam.apache.org/job/beam_PostCommit_Java_DataflowV2_PR/76/

Screen Shot 2022-01-05 at 12 50 12 PM

(your links are for the December 29th's invocations)

**Comment by lukecwik** [comment]
I see, got confused due to the rename of the jenkins job.

**Pull request merged**